plonky3/src/circuit_builder.rs

```diff
@@ -418,13 +418,13 @@ where
             );
 
         // public constraints
-        let public_vals_by_id = self
+        let public_vals_by_name = self
             .constraint_system
             .publics_by_stage
             .iter()
             .zip_eq(public_input_values_by_stage)
             .flat_map(|(publics, values)| publics.iter().zip_eq(values.iter()))
-            .map(|((id, _, _), pi)| (id, *pi))
+            .map(|((name, _, _, _), pi)| (name, *pi))
```

Comment:
This is where the issue was, we were addressing publics by the column name, not the public name